backend-config/namespace_config.go

```diff
@@ -139,10 +140,16 @@ func (nc *namespaceConfig) getFromAPI(ctx context.Context) (map[string]ConfigT,
 		if workspace == nil { // this workspace was not updated, populate it with the previous config
 			previousConfig, ok := nc.workspacesConfig[workspaceID]
 			if !ok {
-				panic(fmt.Errorf(
-					"workspace %q was not updated but was not present in previous config: %+v",
-					workspaceID, req,
-				))
+				nc.logger.Errorw(
+					"workspace was not updated but was not present in previous config",
+					"workspaceID", workspaceID,
+					"req", req,
+				)
+				// reset state here
+				nc.lastUpdatedAt = time.Time{}
```

Comment:
`namespaceConfig#Get` can react on the `ErrIncrementalUpdateFailed `by resetting the `lastUpdatedAt` and calling `getFromAPI` again. This way we won't have to wait for the next poll to happen